File: api-android/src/main/java/com/engarnet/timetree/model/TUser.kt
package com.engarnet.timetree.model

import android.net.Uri
import com.engarnet.timetree.api.v1.api.calendars.response.CalendarMembersResponse
import com.engarnet.timetree.api.v1.api.user.response.UserResponse

data class TUser(
    val id: String,
    val name: String,
    val description: String,
    val imageUrl: Uri?
)

internal fun UserResponse.toModel(): TUser {
    return TUser(
        id = data.id,
        name = data.attributes.name,
        description = data.attributes.description,
        imageUrl = data.attributes.imageUrl?.let { Uri.parse(it) }
    )
}

internal fun CalendarMembersResponse.toModel(): List<TUser> {
    return data.map { user ->
        TUser(
            id = user.id,
            name = user.attributes.name,
            description = user.attributes.description,
            imageUrl = user.attributes.imageUrl?.let { Uri.parse(it) }
        )
    }
}

File: api-ios/TimeTreeAPI/Sources/Model/TUser.swift
//
//  TUser.swift
//  TimeTreeAPI
//
//  Created by EngarNet on 2020/05/02.
//  Copyright © 2020 EngarNet. All rights reserved.
//

import Foundation
import TimeTreeAPICommon

public struct TUser {
    public let id: String
    public let name: String
    public let description: String
    public let imageUrl: URL?
}

extension UserResponse {
    func toModel() -> TUser {
        return TUser(
            id: data.id,
            name: data.attributes.name,
            description: data.attributes.component2(),
            imageUrl: URL(string: data.attributes.imageUrl ?? "")
        )
    }
}

extension CalendarMembersResponse {
    func toModel() -> [TUser] {
        data.map {
            TUser(
                id: $0.id,
                name: $0.attributes.name,
                description: $0.attributes.component2(),
                imageUrl: URL(string: $0.attributes.imageUrl ?? "")
            )
        }
    }
}
